go/runtime/bundle/registry.go

```diff
@@ -202,6 +210,90 @@ func (r *registry) AddBundle(path string, manifestHash hash.Hash) error {
 	return nil
 }
 
+// CleanStaleBundles implements Registry.
+func (r *registry) CleanStaleBundles(ctx context.Context, runtimeID common.Namespace, active version.Version) {
+	for _, v := range r.GetVersions(runtimeID) {
+		select {
+		case <-ctx.Done():
+			return
+		default:
+			if v.Less(active) {
+				r.logger.Info("Removing bundle with version lower then active",
+					"runtimeId", runtimeID,
+					"version", v,
+					"activeVersion", active,
+				)
+				r.cleanBundle(runtimeID, v)
+			}
+		}
+	}
+}
+
+// cleanBundle removes (regular) bundle from the registry.
+//
+// Additionally, it removes exploded subdir and bundle file from the bundle dir.
+func (r *registry) cleanBundle(runtimeID common.Namespace, version version.Version) {
+	explDir, err := r.removeBundle(runtimeID, version)
+	if err != nil {
+		r.logger.Warn("Bundle clean-up unsuccessful: invalid registry state",
+			"error", err,
+			"id", runtimeID,
+			"runtime_version", version,
+		)
+	}
+
+	if err := os.RemoveAll(explDir); err != nil {
+		r.logger.Error("failed to remove stale exploded (regular) bundle dir",
```

Comment:
These parameters are strange:  `runtime_version`, `ExplodedDataDir`. The first one is not consistent with `version` from above, the other has not the same format.